### DISCUSSION TITLE
Data get dropped when ever I restart TimBase

### DISCUSSION THREAD

**Discussion by ankitsoin** [finos/TimeBase-CE#250] [Q&A]
I was changing the data type of a stream field from INTEGER to VARCHAR using MODIFY with CONFIRM CONVERT_DATA. 
TimeBase got stuck and shell got hanged. APIs were failing for ALL streams.
Note: Stream has 2 spaces with each 500mb of data( TimeBase memory)

So I killed all the process ids. Then restarted TimeBase. ALL stream were present, but I lost my few most recent spaces(loaded 3 4 days ago) for multiple streams.

Can you please let me know is it expected/known behavior of TimeBase? What could be possible way out of retaining the data and why this is happenning?
Is it not possible to change the stream field data type with that much size of data?

**Comment by alex-karpovich** [comment]
@ankitsoin 

changing INTEGER to VARCHAR may take time - TimeBase will rewrite all messages (since we have strict schema for each message)
So converting ALL streams may take time, especially when you have gigabytes in storage. 

You will have to wait, progress only visible in Web Administrator.

Best recommendation for you - do not kill TimeBase, unless you cannot connect to it.
use TimeBase Administrator or TimeBase Monitor to check status.
In case of  abnormal termination, TimeBase may loose some recent data

Could you please send me TimeBase logs to see what was happened?

**Reply by ankitsoin** [reply]
Thanks,
Surely, I will provide once I get the same issue again
